test/k8sT/Services.go

```diff
@@ -378,14 +373,13 @@ var _ = Describe("K8sServicesTest", func() {
 		}
 
 		doRequests := func(url string, count int, fromPod string) {
-			By("Making %d curl requests from %s to %q", count, fromPod, url)
-			for i := 1; i <= count; i++ {
-				res, err := kubectl.ExecInHostNetNS(context.TODO(), fromPod, helpers.CurlFail(url))
-				ExpectWithOffset(1, err).To(BeNil(), "Cannot run curl in host netns")
-				ExpectWithOffset(1, res).Should(helpers.CMDSuccess(),
-					"%s host can not connect to service %q (failed in request %d/%d)",
-					fromPod, url, i, count)
-			}
+			By("Making %d curl requests from pod (host netns) %s to %q", count, fromPod, url)
+			cmd := fmt.Sprintf(`/bin/sh -c 'for i in $(seq 1 %d); do %s; done'`, count,
```

Comment:
Holy mother of nesting. But I guess there is no way around this (unless we want to also parallelize curl calls?).